### PR TITLE
Remove basic_plot from Map.peek

### PR DESCRIPTION
**Pull request by Cadair**
:fire: :fire:

### CODE REVIEW

**Comment by nabobalis** [comment]
WE HAD A TEST FOR THIS?

**Comment by Cadair** [comment]
oh yeah so we did :/

File: examples/map/plot_frameless_image.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import matplotlib.pyplot as plt
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+##############################################################################
+# Create a Map
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+
```
Comment (nabobalis) [inline examples/map/plot_frameless_image.py]:
```suggestion
```

File: examples/map/plot_frameless_image.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+##############################################################################
+# Plot the Map without a frame
```
Comment (nabobalis) [inline examples/map/plot_frameless_image.py]:
```suggestion
# Plot the Map without a frame.
```

File: examples/map/plot_frameless_image.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+##############################################################################
+# Start by importing the necessary modules.
+import astropy.units as u
+import matplotlib.pyplot as plt
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+##############################################################################
+# Create a Map
```
Comment (nabobalis) [inline examples/map/plot_frameless_image.py]:
```suggestion
# Create a `sunpy.map.GenericMap`.
```

Maybe?

File: examples/map/plot_frameless_image.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+figure.add_axes(axes)
+
+# Plot the map without any annotations
+# This might raise a warning about the axes being wrong but we can ignore this
+# as we are not plotting any axes.
+im = smap.plot(axes=axes, annotate=False)
+
+##############################################################################
+# At this point you could save the figure with `plt.savefig()` or show it:
+plt.show()
```
Comment (nabobalis) [inline examples/map/plot_frameless_image.py]:
```suggestion

plt.show()
```

File: examples/map/plot_frameless_image.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+axes.set_axis_off()
+figure.add_axes(axes)
+
+# Plot the map without any annotations
+# This might raise a warning about the axes being wrong but we can ignore this
+# as we are not plotting any axes.
+im = smap.plot(axes=axes, annotate=False)
+
+##############################################################################
+# At this point you could save the figure with `plt.savefig()` or show it:
```
Comment (nabobalis) [inline examples/map/plot_frameless_image.py]:
```suggestion
# At this point you could save the figure with ``plt.savefig()`` or show it:
```

File: sunpy/map/mapbase.py [hunk 6]
```diff
@@ -1695,24 +1679,19 @@ def plot(self, annotate=True, axes=None, title=True,
...
-                              " Plots may have unexpected behaviour. To fix this pass "
-                              "'projection=map' when creating the axes",
+        if not wcsaxes_compat.is_wcsaxes(axes):
+            warnings.warn("WCSAxes not being used as the axes object for this plot."
+                          " Plots may have unexpected behaviour. To fix this pass "
+                          "'projection=map' when creating the axes",
+                          SunpyUserWarning)
+            # Check if the image is properly oriented
+            if not np.array_equal(self.rotation_matrix, np.identity(2)):
+                warnings.warn("The axes of this map are not aligned to the pixel grid. Plot axes may be incorrect.",
```
Comment (nabobalis) [inline sunpy/map/mapbase.py]: Stupid question. Were these changes not in a different PR?

Comment (Cadair) [reply sunpy/map/mapbase.py]: yes, but they were inside a `if basic_plot` block, they no longer are.

Comment (nabobalis) [reply sunpy/map/mapbase.py]: OH I'M JUST BLIND AND DIDN'T NOTICE, CARRY ON.

**Pull request merged**